solidity/random-beacon/tasks/staking.ts

```diff
@@ -0,0 +1,110 @@
+import { task, types } from "hardhat/config"
+
+import type { HardhatRuntimeEnvironment } from "hardhat/types"
+
+task(
+  "staking",
+  "Stakes T tokens, increases authorization and registers the operator "
+)
+  .addOptionalParam("owner", "Staking owner address", undefined, types.string)
+  .addOptionalParam("provider", "Staking provider", undefined, types.string)
+  .addOptionalParam("operator", "Staking operator", undefined, types.string)
+  .addOptionalParam("amount", "Staking amount", 1_000_000, types.int)
+  .addOptionalParam(
+    "authorization",
+    "Authorization amount",
+    undefined,
+    types.int
+  )
+  .setAction(async (args, hre) => {
+    await setup(hre, args)
+  })
+
+async function setup(
+  hre: HardhatRuntimeEnvironment,
+  args: {
+    owner: string
+    provider: string
+    operator: string
+    amount: BigInteger
+    authorization: BigInteger
+  }
+) {
+  const { ethers, helpers, getNamedAccounts } = hre
+  const { owner, provider, operator, amount, authorization } = args
+
+  const { deployer } = await getNamedAccounts()
+  const { to1e18 } = helpers.number
+  const t = await helpers.contracts.getContract("T")
+  const staking = await helpers.contracts.getContract("TokenStaking")
+  const randomBeacon = await helpers.contracts.getContract("RandomBeacon")
+  const accounts = await new ethers.providers.JsonRpcProvider().listAccounts()
+  const stakeOwner = owner || accounts[1]
+  const stakingProvider = provider || accounts[2]
+  const stakingOperator = operator || accounts[3]
+  const beneficiary = stakeOwner
+  const authorizer = stakeOwner
+
+  const stakedAmount = to1e18(amount)
+
+  const deployerSigner = await ethers.getSigner(deployer)
```

Comment:
We could pull the deployer as signer here. 
```suggestion
  const { deployer } = await helpers.signers.getNamedSigners())
```

With this change we could remove `const { deployer } = await getNamedAccounts()` above.

Reply:
done